Sources/StreamChatUI/ChatChannel/StreamModalTransitioningDelegate.swift

```diff
@@ -0,0 +1,166 @@
+//
+// Copyright © 2022 Stream.io Inc. All rights reserved.
+//
+
+import UIKit
+
+/// Stream's custom transitioning delegate to handle a modal presentation.
+///
+/// This should be used if you are presenting the ``ChatChannelVC`` in a modal.
+/// The reason this custom transition should be used instead of the native one is because we use an inverted
+/// `UITableView` for the message list component which it doesn't play well with the native modal transition.
+open class StreamModalTransitioningDelegate: NSObject, UIViewControllerTransitioningDelegate {
+    public func presentationController(
+        forPresented presented: UIViewController,
+        presenting: UIViewController?,
+        source: UIViewController
+    ) -> UIPresentationController? {
+        StreamModalPresentationController(presentedViewController: presented, presenting: presenting)
+    }
+}
+
+/// Stream's custom `UIPresentationController` to handle a modal presentation.
+open class StreamModalPresentationController: UIPresentationController {
+    /// The overlay view that is rendered between the presenting and presented view.
+    open var overlayView: UIView = UIView()
+
+    /// The overlay opacity value when the presented view is shown.
+    open var overlayViewAlpha: CGFloat = 0.7
+
+    /// The corner radius of the presented view when it is shown.
+    open var cornerRadius: CGFloat = 22
+
+    /// The presented view height scale.
+    open var presentedViewHeightScale: CGFloat = 0.94
+
+    /// The presenting view height scale when the presented view is shown.
+    open var presentingViewHeightScale: CGFloat = 0.9
+
+    override public init(presentedViewController: UIViewController, presenting presentingViewController: UIViewController?) {
+        super.init(presentedViewController: presentedViewController, presenting: presentingViewController)
+        setup()
+    }
+
+    /// Setup the overlay view and gesture recognizers of the `UIPresentationController`.
+    open func setup() {
+        overlayView.backgroundColor = UIColor.black
+        overlayView.autoresizingMask = [.flexibleWidth, .flexibleHeight]
+        overlayView.isUserInteractionEnabled = true
```

Comment:
The default value of this property is true.

Reply:
✅